### PR TITLE
test(unit): angular v12 example app (the tour of heroes)

### PR DESCRIPTION
**Pull request by rfprod**
## Summary

- [x] add [the tour of heroes](https://angular.io/generated/live-examples/testing/stackblitz) app and specs to the example angular v12 app;
- [x] reuse sources and tests recently added to the v11 example app;
- [x] remove sources that are not present in the tour of heroes app;
- [x] enable strict compiler mode; 
- [x] fix tsconfig - there's no more `services` dir, map the `shared` dir instead;
- [x] remove tslint comments;
- [x] use mockResolvedValue instead of toPromise;
- [x] replace the done callback, jest 27 default runner does not have it any more;

## Does this PR introduce a breaking change?

- [X] No

## Other information

- [x] verify tests workability locally:
```bash
yarn test
yarn test-isolated
yarn test-esm
yarn test-esm-isolated
```

### CODE REVIEW

**Comment by rfprod** [comment]
@ahnpnl review please.

File: examples/example-app-v12/src/app/demo/async-helper.spec.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
+import { fakeAsync, tick, waitForAsync } from '@angular/core/testing';
+import { jest } from '@jest/globals';
+import { DoneFn } from '@jest/types/build/Circus';
```
Comment (ahnpnl) [inline examples/example-app-v12/src/app/demo/async-helper.spec.ts:3]: we should not import from `build` which is not recommended by Jest team. Would you please also correct other places?

File: examples/example-app-v12/src/app/hero/hero-detail.component.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/* tslint:disable:member-ordering */
```
Comment (ahnpnl) [inline examples/example-app-v12/src/app/hero/hero-detail.component.ts:1]: this comment should be removed. Would you please also correct any other places where tslint comments appear?

**Comment by rfprod** [comment]
@ahnpnl will update the PR soon. I'll update other open PRs accordingly

**Comment by rfprod** [comment]
@ahnpnl all done, check again please.

File: examples/example-app-v12/src/app/demo/demo.spec.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    beforeEach(() => {
+      service = new ValueService();
+    });
+
+    it('#getValue should return real value', () => {
+      expect(service.getValue()).toBe('real value');
+    });
+
+    // eslint-disable-next-line jest/no-done-callback
+    it('#getObservableValue should return value from observable', (done: jest.DoneCallback) => {
```
Comment (ahnpnl) [inline examples/example-app-v12/src/app/demo/demo.spec.ts:22]: `jest-circus` runner (which is default in Jest 27) doesn't have `done` callback anymore :)

You can convert the observable to `Promise` and use `async/await` to replace `done` callback. Not sure why the tests use `done` callback didn't fail. We should remove the `done` callback for other places too.

Comment (rfprod) [reply examples/example-app-v12/src/app/demo/demo.spec.ts:22]: k, will do soon

Comment (rfprod) [reply examples/example-app-v12/src/app/demo/demo.spec.ts]: @ahnpnl resolved

**Review by ahnpnl — APPROVED** [review]

**Comment by ahnpnl** [comment]
Thanks!!

**Pull request merged**